scanner/library.go

```diff
@@ -1,20 +1,26 @@
 package scanner
 
 import (
-	"github.com/aquasecurity/trivy/pkg/fanal/types"
+	ftypes "github.com/aquasecurity/trivy/pkg/fanal/types"
+	"github.com/aquasecurity/trivy/pkg/purl"
+	"github.com/aquasecurity/trivy/pkg/types"
+	"github.com/future-architect/vuls/logging"
 	"github.com/future-architect/vuls/models"
+	"github.com/package-url/packageurl-go"
 )
 
-func convertLibWithScanner(apps []types.Application) ([]models.LibraryScanner, error) {
+func convertLibWithScanner(apps []ftypes.Application) ([]models.LibraryScanner, error) {
 	scanners := []models.LibraryScanner{}
 	for _, app := range apps {
 		libs := []models.Library{}
 		for _, lib := range app.Libraries {
+			purl := newPURL(app.Type, types.Metadata{}, lib)
 			libs = append(libs, models.Library{
-				Name:     lib.Name,
-				Version:  lib.Version,
-				FilePath: lib.FilePath,
-				Digest:   string(lib.Digest),
+				Name:       lib.Name,
+				Version:    lib.Version,
+				PackageURL: purl.ToString(),
```

Comment:
When purl is nil, won't purl.ToString() panic?

Reply:
A test case like this will cause panic.
```golang
 
package scanner

import (
	"reflect"
	"testing"

	ftypes "github.com/aquasecurity/trivy/pkg/fanal/types"
	"github.com/future-architect/vuls/models"
)

func Test_convertLibWithScanner(t *testing.T) {
	tests := []struct {
		name    string
		args    []ftypes.Application
		want    []models.LibraryScanner
		wantErr bool
	}{
		{
			name: "test",
			args: []ftypes.Application{
				{
					Type:     ftypes.GoModule,
					FilePath: "go.mod",
					Libraries: ftypes.Packages{
						{
							Name: "",
						},
					},
				},
			},
			want: []models.LibraryScanner{},
		},
	}
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			got, err := convertLibWithScanner(tt.args)
			if (err != nil) != tt.wantErr {
				t.Errorf("convertLibWithScanner() error = %v, wantErr %v", err, tt.wantErr)
				return
			}
			if !reflect.DeepEqual(got, tt.want) {
				t.Errorf("convertLibWithScanner() = %v, want %v", got, tt.want)
			}
		})
	}
}
```

Reply:
If this diff can help with your corrections...
```diff
:100644 100644 e62373d 0000000 M	scanner/library.go

diff --git a/scanner/library.go b/scanner/library.go
index e62373d..097def1 100644
--- a/scanner/library.go
+++ b/scanner/library.go
@@ -6,7 +6,6 @@ import (
 	"github.com/aquasecurity/trivy/pkg/types"
 	"github.com/future-architect/vuls/logging"
 	"github.com/future-architect/vuls/models"
-	"github.com/package-url/packageurl-go"
 )
 
 func convertLibWithScanner(apps []ftypes.Application) ([]models.LibraryScanner, error) {
@@ -14,11 +13,10 @@ func convertLibWithScanner(apps []ftypes.Application) ([]models.LibraryScanner,
 	for _, app := range apps {
 		libs := []models.Library{}
 		for _, lib := range app.Libraries {
-			purl := newPURL(app.Type, types.Metadata{}, lib)
 			libs = append(libs, models.Library{
 				Name:       lib.Name,
 				Version:    lib.Version,
-				PackageURL: purl.ToString(),
+				PackageURL: newPURL(app.Type, types.Metadata{}, lib),
 				FilePath:   lib.FilePath,
 				Digest:     string(lib.Digest),
 			})
@@ -32,11 +30,14 @@ func convertLibWithScanner(apps []ftypes.Application) ([]models.LibraryScanner,
 	return scanners, nil
 }
 
-func newPURL(pkgType ftypes.TargetType, metadata types.Metadata, pkg ftypes.Package) *packageurl.PackageURL {
+func newPURL(pkgType ftypes.TargetType, metadata types.Metadata, pkg ftypes.Package) string {
 	p, err := purl.New(pkgType, metadata, pkg)
 	if err != nil {
 		logging.Log.Errorf("Failed to create PackageURL: %+v", err)
-		return nil
+		return ""
 	}
-	return p.Unwrap()
+	if p == nil {
+		return ""
+	}
+	return p.Unwrap().ToString()
 }
```